### ISSUE TITLE
When playback is paused, script CPU usage increases and listing/cleaning log messages are rapidly produced

### ISSUE THREAD

**Issue by Spiritreader** [RTUnit/Jellyfin-Transcodes-cleanup#11]
Whenever a transcoding session is active and a user pauses the playback, I get an increased number of

```
Listing/cleaning TS files sequentially after the LA file
```
messages in the log.

This logging event will happen every 0.2 seconds until the playback is terminated or resumed, which results in quite a large log over time (9.4MB in 24 hours)

CPU usage of the script then increases and stays there for the remainder. 
It's mostly on one core, but this is a grafana screenshot of the jellyfin container usage. So what you see is total CPU usage out of all cores.
![image](https://github.com/RTUnit/Jellyfin-Transcodes-cleanup/assets/18438899/7aba023e-aaf0-4396-86af-b9b75fee5583)

Bufmon says this:

```
=== 6361c6bcff2968e4b703772cae1118c1 ============================ v1.1
2085  2.96 MB    ► playback                            POS:        220
2086  3.06 MB  ■■■ served                             PID:14151 KILLED
2087  3.07 MB                                        STATE:    RUNNING
2088  3.06 MB
2089  3.21 MB
...
2148  3.07 MB
2149  3.07 MB
2150  1.03 MB
COUNT: 66                                  PERFORMANCE: ᴑ loading...
----------------------------------------------------------------------
```

Log output
```
I 2024-01-17 00:32:16.568678049 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially after the LA file (TS ID > 2086):
I 2024-01-17 00:32:16.815724641 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially before the LA file (TS ID < 2086):
I 2024-01-17 00:32:16.860201121 [28386]    Keeping (LA-1): /transcode/6361c6bcff2968e4b703772cae1118c12085.ts
I 2024-01-17 00:32:16.861745438 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially after the LA file (TS ID > 2086):
I 2024-01-17 00:32:17.108499687 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially before the LA file (TS ID < 2086):
I 2024-01-17 00:32:17.155138787 [28386]    Keeping (LA-1): /transcode/6361c6bcff2968e4b703772cae1118c12085.ts
I 2024-01-17 00:32:17.156845496 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially after the LA file (TS ID > 2086):
I 2024-01-17 00:32:17.405699258 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially before the LA file (TS ID < 2086):
I 2024-01-17 00:32:17.449101214 [28386]    Keeping (LA-1): /transcode/6361c6bcff2968e4b703772cae1118c12085.ts
I 2024-01-17 00:32:17.450550692 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially after the LA file (TS ID > 2086):
I 2024-01-17 00:32:17.699379865 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially before the LA file (TS ID < 2086):
I 2024-01-17 00:32:17.742814814 [28386]    Keeping (LA-1): /transcode/6361c6bcff2968e4b703772cae1118c12085.ts
I 2024-01-17 00:32:17.744279560 [28386] [6361c6bcff2968e4b703772cae1118c1] Listing/cleaning TS files sequentially after the LA file (TS ID > 2086)
```

In my case playback often gets paused and resumed hours later, which I why I noticed it in the first place.
Not sure if finding this is noteworthy or useful, but I encountered that behavior and thought of it as odd that the script busy-waits with more CPU usage than when playback is actually happening.

Some specs:

Jellyfin 10.8.13 running as docker container on Unraid.
I did not modify any of the global variables in the configuration files.

**Comment by RTUnit** [comment]
That peak we observe in grafana is when playback is paused?

Could be that some shell command is more resource hungry... would be interesting to find out which it is. One quick remedy could be setting sleep at the end of the script to a larger value, for example set it to 1 second:
```
sleep 1
```
This should give more rest to the CPU and not affect much the performance of the script. I had this value larger in beginning, but decreased it due to shell commands were extending the time of loop cycle.

If it helps to reduce your CPU load during pause then I am considering to add a feature to the script to have a variable `sleep` value for controlling the minimum time for each cycle of the main loop.

**Comment by Spiritreader** [comment]
> That peak we observe in grafana is when playback is paused?
> 
> Could be that some shell command is more resource hungry... would be interesting to find out which it is. One quick remedy could be setting sleep at the end of the script to a larger value, for example set it to 1 second:
> 
> ```
> sleep 1
> ```
> 
> This should give more rest to the CPU and not affect much the performance of the script. I had this value larger in beginning, but decreased it due to shell commands were extending the time of loop cycle.
> 
> If it helps to reduce your CPU load during pause then I am considering to add a feature to the script to have a variable `sleep` value for controlling the minimum time for each cycle of the main loop.

no, the peak is when I restarted jellyfin. The constant CPU usage before that is when playback was paused.

I will try adjusting the sleep parameter!

Update: Changing the sleep to 1 second did indeed reduce the CPU usage during paused playback, by, well, a factor of 10! 😄 
After some further testing we've not set the sleep to 300 seconds, which results in barely any increased CPU usage.
